app/views/layouts/application.html.haml

```diff
@@ -1,3 +1,28 @@
+%nav.navbar.navbar-inverse{:role => "navigation"}
+  .container-fluid
+    / Brand and toggle get grouped for better mobile display
+    .navbar-header
+      %button.navbar-toggle{"data-target" => "#bs-example-navbar-collapse-1", "data-toggle" => "collapse", :type => "button"}
+        %span.sr-only Toggle navigation
+        %span.icon-bar
+        %span.icon-bar
+        %span.icon-bar
+      %a.navbar-brand{:href => "#"} Defcon
+    / Collect the nav links, forms, and other content for toggling
+    #bs-example-navbar-collapse-1.collapse.navbar-collapse
+      %ul.nav.navbar-nav
+        %li
+          %a{:href => "#"} Link
+    %ul.nav.navbar-nav.navbar-right
+    - if user_signed_in?
+      %li= link_to 'Sign out', destroy_user_session_path, :method => :delete
+    - else
+      %li= link_to 'Sign in', new_user_session_path
+    %li
+      %a{:href => "#"} Link
+/ /.navbar-collapse
+/ /.container-fluid
+
```

Comment:
Nice work getting the navbar in here.

In HAML `!!!` renders the doc-type of the HTML document, which basically says which version of HTML the browser should use (browsers are pretty smart and autodetect and figure stuff out if it's missing).  You can see the details here.

http://www.w3schools.com/tags/tag_doctype.asp

So since that needs to be at the top, this should actually be nested a bit lower.